internal/client/github.go

```diff
@@ -449,6 +449,11 @@ func (c *githubClient) createOrUpdateRelease(ctx *context.Context, data *github.
 			ctx.Config.Release.GitHub.Name,
 			data,
 		)
+		if err != nil {
+			log.WithField("name", data.GetName()).
+				Debug("release creation failed")
+			return nil, err
+		}
```

Comment:
maybe something like this:

```suggestion
		if resp == nil {
			log.WithField("name", data.GetName()).
				WithField("release-id", release.GetID()).
				Debug("release creation failed")
			return nil, err
		}
```

Reply:
Why do you prefer to check on a nil response instead of the error? I'm asking since CreateRelease responds also with a nil release struct (see: https://github.com/google/go-github/blob/c96ef954c31ce1a9f74675b8ec2c66e24292ea51/github/repos_releases.go#L221) so release.GetID() will either be empty or error as well.

Reply:
not necessarily, it returns nil if creating the request fails, but it returns a response if the request fails

that said, we should probably add a `WithError(err)` to that log as well

Reply:
Ah, thanks! I've updated it to cover both cases. I removed the `WithField("release-id", release.GetID())` because release is nil in both cases.